apps/crn-frontend/src/network/teams/TeamOutput.tsx

```diff
@@ -56,28 +62,67 @@ export function paramOutputDocumentTypeToResearchOutputDocumentType(
 
 type TeamOutputProps = {
   teamId: string;
+  researchOutputData?: ResearchOutputResponse;
 };
-const TeamOutput: React.FC<TeamOutputProps> = ({ teamId }) => {
+const TeamOutput: React.FC<TeamOutputProps> = ({
+  teamId,
+  researchOutputData,
+}) => {
   const paramOutputDocumentType = useParamOutputDocumentType(teamId);
-  const documentType = paramOutputDocumentTypeToResearchOutputDocumentType(
-    paramOutputDocumentType,
-  );
+  const documentType =
+    researchOutputData?.documentType ||
+    paramOutputDocumentTypeToResearchOutputDocumentType(
+      paramOutputDocumentType,
+    );
   const team = useTeamById(teamId);
   const [errors, setErrors] = useState<ValidationErrorResponse['data']>([]);
 
   const { canCreateUpdate } = useContext(ResearchOutputPermissionsContext);
 
   const createResearchOutput = usePostTeamResearchOutput();
+  const updateResearchOutput = usePutTeamResearchOutput(
+    researchOutputData?.id as string,
+  );
 
   const getLabSuggestions = useLabSuggestions();
   const getAuthorSuggestions = useAuthorSuggestions();
   const getTeamSuggestions = useTeamSuggestions();
   const researchTags = useResearchTags();
 
+  const handleSave = (output: ResearchOutputPostRequest) =>
+    createResearchOutput(output).catch((error: unknown) => {
+      if (error instanceof BackendError) {
+        const { response } = error;
+        if (
+          isValidationErrorResponse(response) &&
+          validationErrorsAreSupported(response, ['/link', '/title'])
+        ) {
+          setErrors(response.data);
+          return;
+        }
+      }
+      throw error;
+    });
+
+  const handleUpdate = (output: ResearchOutputPutRequest) =>
```

Comment:
Rather than duplicate the catch function twice could we just abstract that instead of these two new functions?